### PR TITLE
[Docs] Add Python dependencies management docs

### PR DESCRIPTION
**Pull request by p-wysocki**
### Details:
 - Add docs explaining Python dependencies management using `constraints.txt` files
 - Related PR https://github.com/openvinotoolkit/openvino/pull/15343

### Tickets:
 - 103516

### CODE REVIEW

**Review by jiwaszki — APPROVED** [review]

File: src/bindings/python/docs/requirements_management.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# Python requirements and version constraints management
+
+OpenVINO uses a pip built-in feature called constraints in order to reduce requirements complexity.
```
Comment (kblaszczak-intel) [inline src/bindings/python/docs/requirements_management.md:3]:
```suggestion
OpenVINO uses a pip built-in feature called "constraints" in order to reduce the complexity of requirements. 
```

File: src/bindings/python/docs/requirements_management.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
+# Python requirements and version constraints management
+
+OpenVINO uses a pip built-in feature called constraints in order to reduce requirements complexity. 
+
+## What are constraints files?
+
+Official constraints [documentation](https://pip.pypa.io/en/stable/user_guide/#constraints-files) describes them as:
+> Constraints files are requirements files that only control which version of a requirement is installed, not whether it is installed or not. Their syntax and contents is a subset of Requirements Files, with several kinds of syntax not allowed: constraints must have a name, they cannot be editable, and they cannot specify extras. In terms of semantics, there is one key difference: Including a package in a constraints file does not trigger installation of the package.
+
+Constraints files have `.txt` extension and are an addon to regular `requirements.txt` files. They are useful when the project has multiple components, each with their own dependencies.
```
Comment (kblaszczak-intel) [inline src/bindings/python/docs/requirements_management.md:10]:
```suggestion
Constraints files have the `.txt` extension and are addons to regular `requirements.txt` files. They are useful when the project has multiple components, each with its own dependencies.
```

File: src/bindings/python/docs/requirements_management.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+Thanks to constraints files we can specify only the package names for different components and keep their versions in a centralized constraints file, which is significanly easier to maintain. 
+
+
+### Example
+Two requirements files are linked to a centralized constraints file using `-c` flag.
```
Comment (kblaszczak-intel) [inline src/bindings/python/docs/requirements_management.md:16]:
```suggestion
Two requirements files are linked to a centralized constraints file using the `-c` flag.
```

File: src/bindings/python/docs/requirements_management.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+Placing [environment markers](https://peps.python.org/pep-0508/) in `constraints.txt` files can be prone to bugs due to how pip resolver interprets them. Markers should be placed only in `requirements.txt` files unless it is certain they will work as intended.
+
+#### 2. Inconsistent version requirements
+If a package version differs between `requirements.txt` files, it can't be unified in the common `constraints.txt` file. The workarounds, in order of preference, are:
+- Align the package versions between `requirements.txt` files
+- Exclude this package from `constraints.txt` and keep its version in `requirements.txt`
+- Exclude this `requirements.txt` file from constraints system
+
+#### 3. Lack of support in distutils
+`distutils` is a package used for building Python wheels. It does not offer support for constraints files, so custom requirements parsers had to be added. An example parser can be found [here](https://github.com/openvinotoolkit/openvino/blob/master/tools/mo/setup.py) - functions `read_constraints()` and `read_requirements()`.
```
Comment (kblaszczak-intel) [inline src/bindings/python/docs/requirements_management.md:75]:
```suggestion
`distutils` is a package used for building Python wheels. It does not offer support for constraints files, so custom requirement parsers had to be added. A parser example can be found [here](https://github.com/openvinotoolkit/openvino/blob/master/tools/mo/setup.py) - functions `read_constraints()` and `read_requirements()`.
```

File: src/bindings/python/docs/requirements_management.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+## Implementation in OpenVINO
+The implementation in OpenVINO is a subject to change. At the time of writing there are three `constraints.txt` files with the following requirements coverage:
```
Comment (kblaszczak-intel) [inline src/bindings/python/docs/requirements_management.md:78]:
```suggestion
The implementation in OpenVINO is a subject to change. At the time of writing, there are three `constraints.txt` files with the following requirement coverage:
```

**Review by kblaszczak-intel — APPROVED** [review]
Just minor adjustments proposed

**Pull request merged**